### ISSUE TITLE
Weird bug with $lte after migration to 6.27.0

### ISSUE THREAD

**Issue by ellenaua** [sequelize/sequelize#15463]
## Issue Creation Checklist

- [ ] I understand that my issue will be automatically closed if I don't fill in the requested information
- [ ] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

We were using Sequelize 4.44 and had a query which looks this way:

```
const transactions = await db.Transaction.findAll({
    where: {
        createdAt: {
            $gte: '2022-11-06'
         }
    }
})
```

It was selecting from database all transactions having "createdAt" date >= given date 
`WHERE "transaction"."createdAt" >= '2022-11-06 00:00:00.000 +00:00'`

 **Then I migrated repo to Sequelize 6.27.0 BUT I forgot to change $gte to Op.gte in this query**. 

And I noticed that instead of thousands of transactions this query started to return 0 records all the time.
During debugging, I found that query which runs under the hood is:
`WHERE "transaction"."createdAt" = '2022-12-15 22:00:00.000 +00:00'` 

This is extremely strange because 2022-12-15 it's the day which was yesterday.
It's even not the current date.

### Reproducible Example

Here is the link to the SSCCE for this issue:

### What do you expect to happen?

When Sequelize 6 gets old-style query like `createdAt: {  $gte: '2022-11-06' }`

it shoud do one of the following:

- use "Invalid date" in a query
- throw an error
- log a warning for developer that this format is not supported anymore

### What is actually happening?

When Sequelize 6 gets old-style query like `createdAt: {  $gte: '2022-11-06' }`

it generates a query  `WHERE "createdAt" = '2022-12-15 22:00:00.000 +00:00'`  

which is totally incorrect

### Environment

- Sequelize version: sequelize@6.27.0
- Node.js version: v16.15.1
- If TypeScript related: TypeScript version:
- Database & Version: PostgreSQL 11.13 
- Connector library & Version: pg@8.8.0

## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by frostzt** [comment]
Weird so I tried the following:

```typescript
class Foo extends Model {}

Foo.init({
    name: DataTypes.TEXT,
}, {
    sequelize,
    modelName: 'Foo',
    timestamps: true
});

await Foo.findOne({ where: { createdAt: { $gte: '2022-11-06' } } });
```

And the query generated was the following:

```sql
Executed (default): SELECT "id", "name", "createdAt", "updatedAt" FROM "Foos" AS "Foo" WHERE "Foo"."createdAt" = '2022-12-15 18:30:00.000 +00:00' LIMIT 1;
```

I'll try looking into this!

**Comment by ephys** [comment]
I can confirm this bug exists, it's parsing the input as local time then formatting the output as UTC. Same issue if the input is `2022-11-06T00:00:00`, but OK if a timezone offset is specified (`2022-11-06T00:00:00Z`)

Parsing the input as local time definitely sounds like a mistake to me, but I'm worried someone relies on in in v6 and changing it will break something.

I'm investigating whether this is still an issue following the datatype rewrite in Sequelize 7

**Comment by ephys** [comment]
In v7 (following the not-yet-released data types rewrite), `2022-01-01` produces `2022-01-01T00:00:00Z` as expected, but `2022-01-01T00:00:00` is parsed as local time. This is the logic the JS Date constructor uses.

I would argue that inputs like `2022-01-01T00:00:00` should throw an error requesting the user to specify a timezone offset to prevent any ambiguity?

**Comment by ellenaua** [comment]
@ephys but it's not just timezone issue - see my example - I have '2022-11-06' in query but Sequelize uses '2022-12-15' under the hood. These are two totally different dates

**Comment by ephys** [comment]
Yes it seems to have falled back to "now" for some reason, when it should definitely have thrown an error because $gte is not allowed anymore

**Comment by frostzt** [comment]
@ephys that makes sense. Do we already have something to not allow `$gte`? Like for v7 I think we can use?!

**Comment by ephys** [comment]
It's managed by the DataType itself when it tries to parse the input

That is also fixed in v7, I've added the following unit test and test test suites are green:

![image](https://user-images.githubusercontent.com/1280915/208072299-2fe674f8-6f0a-48be-b52d-c1b34e91ffd9.png)

**Comment by frostzt** [comment]
Hey @ephys since this needs a backfix I'll pick this up!

**Comment by ellenaua** [comment]
My further debugging shows that magic happens here
https://github.com/sequelize/sequelize/blob/v6.27.0/src/data-types.js#L471-L477

Object { '$gte': '2022-11-06T00:00:00.000Z' } comes into this function
and then function this._applyTimezone turns it into 2022-12-15 22:00:00.000 +00:00

(by the way I am a maintainer of moment-timezeone 😃 )
